File: Data Structures & Algorithms/max-consecutive-ones/submission-2.rs
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut cnt = 0;
        for &num in &nums {
            cnt = if num == 1 { cnt + 1 } else { 0 };
            res = res.max(cnt);
        }
        res
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: list[int]) -> int:
        dp = [-1] * len(nums)

        def rob(idx: int) -> int:
            if idx >= len(nums):
                return 0
            if dp[idx] != -1:
                return dp[idx]
            dp[idx] = max(nums[idx] + rob(idx + 2), rob(idx + 1))
            return dp[idx]
        
        rob(0)
        return dp[0]